File: codegen/codegen-erlang/src/test/resources/ir/structure_decode_encode_basic_item.expected.erl
decode_basic_item(undefined) -> undefined;
decode_basic_item(null) -> undefined;
decode_basic_item(Map) when is_map(Map) ->
    #basic_item{
        name = maps:get(<<"name">>, Map, undefined),
        count = maps:get(<<"count">>, Map, undefined)
    }.

encode_basic_item(undefined) -> undefined;
encode_basic_item(Record) -> maps:filter(fun (_, V) ->
    V =/= undefined
end, #{<<"name">> => Record#basic_item.name, <<"count">> => Record#basic_item.count}).

File: codegen/codegen-elixir/src/test/resources/ir/structure_decode_encode_basic_item.expected.ex
defp decode_basic_item(nil), do: :nil
defp decode_basic_item(map) when is_map(map) do
  %Types.BasicItem{
    name: Map.get(map, "name"),
    count: Map.get(map, "count")
  }
end

defp encode_basic_item(nil), do: :nil
defp encode_basic_item(record = %Types.BasicItem{}) do
  _map =
    %{
      "name" => record.name,
      "count" => record.count
    }
    |> Enum.reject(fn {_k, v} -> is_nil(v) end)
    |> Map.new()
end